app/src/main/java/org/openobservatory/ooniprobe/common/ReachabilityManager.java

```diff
@@ -61,4 +61,14 @@ public static Boolean isCharging(Context context) {
                 status == BatteryManager.BATTERY_STATUS_FULL;
         return isCharging;
     }
+
+    /* From https://developer.android.com/training/monitoring-device-state/battery-monitoring#DetermineChargeState */
+    public static float getChargingLevel(Context context) {
+        IntentFilter ifilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
+        Intent batteryStatus = context.registerReceiver(null, ifilter);
+        int level = batteryStatus.getIntExtra(BatteryManager.EXTRA_LEVEL, -1);
+        int scale = batteryStatus.getIntExtra(BatteryManager.EXTRA_SCALE, -1);
```

Comment:
If both `getInExtra` return `-1` the `batteryPct` is `-1 * 100 / (float) -1` which is equal to `100`. I think we should dd an explicit case here `if (level < 0 || scale < 0) { return 0; }`.

Reply:
I used the code from google https://developer.android.com/training/monitoring-device-state/battery-monitoring#DetermineChargeState

Reply:
okay, then, we're going to assume this is fine since it comes from a blog post in developer.android.com